### ISSUE TITLE
Can not clear the AutoComplete label

### ISSUE THREAD

**Issue by Janpot** [mui/mui-toolpad#2467]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Steps:

1. Add an autocomplete to the page
2. Go to the `label` property
3. Try to clear the textfield

### Current behavior 😯

The default value "Search..." always appears

### Expected behavior 🤔

Can clear the label of the autocomplete

**Comment by Janpot** [comment]
This seems to happen because clearing the string editor resets the underlying value to `undefined` which makes it adopt the default value again, we should only clear the value in the model if the user enters the default value.

Following changes seem to fix the problem for me:

```diff
diff --git a/packages/toolpad-app/src/toolpad/propertyControls/string.tsx b/packages/toolpad-app/src/toolpad/propertyControls/string.tsx
index 9b7b40ad3..a2c0d0b65 100644
--- a/packages/toolpad-app/src/toolpad/propertyControls/string.tsx
+++ b/packages/toolpad-app/src/toolpad/propertyControls/string.tsx
@@ -6,9 +6,9 @@ import PropertyControl from '../../components/PropertyControl';
 function StringPropEditor({ propType, label, value, onChange, disabled }: EditorProps<string>) {
   const handleChange = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
-      onChange(event.target.value || undefined);
+      onChange(event.target.value === propType.default ? undefined : event.target.value);
     },
-    [onChange],
+    [onChange, propType.default],
   );
 
   return (
```

A test for this could be helpful

**Comment by jyash97** [comment]
Hey @Janpot 

New to the repo, and trying to understand the codebase, but shouldn't the state of ui component prop handle the prop value instead of doing the check for default value , for eg: 


```
label: "Search..."

// user clears

label: ""
```


Shouldn't this handle the case, why the value is getting reset on clear and where does this happen? Trying to figure out the whole flow of component, so any details would be helpful :D

Looking forward to contributing to the repo 😄 

Thanks!

**Comment by Janpot** [comment]
We could also just do

```tsx
onChange(event.target.value)
```

This would work as well, but this would have one difference related to what ends up in the `page.yml` on disk. Mainly it would never return a file to its original state when setting the property back tot he default value in the UI. Not a huge deal though.

**Comment by JerryWu1234** [comment]
![image](https://github.com/mui/mui-toolpad/assets/27261522/dada47cd-5d0f-480c-904f-8e1e5ea381c0)

I can work as well.

@jyash97 Actually, I'm not totally sure what you said.

@Janpot 
I follow your code and add some tests.
That's all pr. Isn't it?

**Comment by Janpot** [comment]
> That's all for this pr. Isn't it?

Yes, no need to write a whole new test fixture, you can just augment [this one](https://github.com/mui/mui-toolpad/blob/master/test/integration/propControls/basic.spec.ts)
